lib/ansible/module_utils/urls.py

```diff
@@ -792,7 +792,7 @@ def open_url(url, data=None, headers=None, method=None, use_proxy=True,
     # FIXME: change the following to use the generic_urlparse function
     #        to remove the indexed references for 'parsed'
     parsed = urlparse(url)
-    if parsed[0] != 'ftp':
+    if parsed[0] != b('ftp'):
```

Comment:
urlparse will return bytes if passed bytes, text strings if passed a text string:
``` python
>>> from urllib.parse import urlparse
>>> urlparse('ftp://ftp.sunsite.unc.edu/')[0]
'ftp'
>>> urlparse(b'ftp://ftp.sunsite.unc.edu/')[0]
b'ftp'
```

So we probably want to find out what's calling this and why it is passing in a byte string on python3.